kernel/src/process.rs

```diff
@@ -1595,13 +1604,13 @@ impl<C: 'static + Chip> Process<'a, C> {
         {
             if config::CONFIG.debug_load_processes {
                 debug!(
-                    "[!] flash=[{:#010X}:{:#010X}] process={:?} - couldn't allocate flash region",
+                    "[!] flash=[{:#010X}:{:#010X}] process={:?} - couldn't allocate MPU region for flash",
                     app_flash.as_ptr() as usize,
                     app_flash.as_ptr() as usize + app_flash.len(),
                     process_name
                 );
             }
-            return Ok((None, 0));
+            return Err(ProcessLoadError::InternalError);
```

Comment:
I'm not sure making this error `InternalError` is right. I think we want to save the internal error for only issues where the kernel is wrong. In this case, however, I think an app loaded with a length in flash that the MPU doesn't expect will cause an error, which is maybe arguably a kernel error, but the fix is to correct the app size, not patch the kernel.

Likely a new error would be better, one that indicates it could be the app or a bad MPU implementation.